db-client-java/src/main/java/com/eventstore/dbclient/GrpcClient.java

```diff
@@ -182,13 +183,26 @@ private boolean discover(UUID previousId, Optional<Endpoint> candidate) {
     private boolean loadServerFeatures() {
         try {
             serverInfo = ServerFeatures.getSupportedFeatures(settings, channel);
+            serverInfoLoaded = true;
             return true;
         } catch (ServerFeatures.RetryableException e) {
             logger.warn("An exception happened when fetching server supported features. Retrying connection attempt.", e);
             return false;
         }
     }
 
+    public ServerVersion getServerVersion() throws RuntimeException {
```

Comment:
You don't have to declare that a function would throw a `RuntimeException`.

Reply:
will fix, thanks

Reply:
fixed